### ISSUE TITLE
heim user accounts

### ISSUE THREAD

**Issue by mastensg** [bitraf/infrastructure#6]
every active bitraf member should have access to heim.

- [ ] automatically create accounts for new members

https://p2k12.bitraf.no/passwd includes every member including those who have stopped paying.

the p2k12 passwd document can be converted into a format suitable for ansible. i have added an example entry in https://github.com/bitraf/infrastructure/blob/master/bite/heim/roles/users/vars/main.yml

- [ ] how set non-paying members' shell to `/bin/false` until they resume paying

**Comment by tingox** [comment]
Should it be automatic? Or should it require  action (a simple action) from the user? Thios way, everyone has the possibility, but only those who actively choose to (via a simple action) will get an account.

**Comment by mastensg** [comment]
@tingox as it is, it does require `login passwd` to be run once in p2k12

is there any reason to require more?

**Comment by tingox** [comment]
I don't have the answer, but there is a difference between:
- when you do this ('passwd login') you get  set a password for the door and you get an account on heim (which you might use or not)
and
- if you want an account on heim you will have to do <action>
the former way might lead to some unused accounts on heim, the latter will ensure that people actively requests an account (it might still end up unused after a while). The question is: are we (Bitraf) happy either way, or do we prefer one solution over the other?

**Comment by odinho** [comment]
Torfinn, difference between:
- passwd door
- passwd login

Only last will give a login. I think that's okay. That is a required action. But it would be nice to time-out the logins. Or maybe just checking if the account is a valid member still is enough.

**Comment by tingox** [comment]
True, I mixed those two up. But the point still stands: Is it ok if members get an account on heim as a "side-effect" of another action? (in this case, setting a wiki password)
Or should it have its own action?
Or are we (Bitraf) happy either way?

**Comment by mastensg** [comment]
i don't see any issue with automatically creating accounts. if they remain unused, they take up almost no resources, and if someone wants to make use of them, then there are no extra steps for them to make

also, as odinho wrote, i have already created all the accounts

**Comment by tingox** [comment]
There is no technical issue in automating this, and I see that it has happened already. For the record, the point I was trying to make (unsuccessfully it seems) is a human "issue" if you can call it that:
- things we (humans) get for free (or automatically) is perceived by ourselves as something with low or no value.
- things we have to pay for (or ask for) is perceived as having a higher value than things we get for free.
Perhaps we (Bitraf) should consider this fact when planning future service for our users.
